### PR TITLE
Refactor quick-equip verb

### PR DESCRIPTION
**Pull request by Lobachevskiy**
Fixes https://github.com/tgstation/-tg-station/issues/12856 with the power of OBJECT ORIENTED PROGRAMMING, so the priority and slots can be redefined for any item. For example, make handcuffs go into security belt before the pockets.

Also allows any mob for which get_active_hand() is defined and returns an item to quick-equip it, provided there are valid slots. Drone quick-equip is less snowflakey. Didn't touch HUD, so it's only accessible via the hotkey for everything but humans (was already the case with drones).

### CODE REVIEW

**Comment by Bawhoppen** [comment]
> For example, make handcuffs go into security belt before the pockets.

should make it handcuff yourself

**Comment by Lo6a4evskiy** [comment]
> should make it handcuff yourself 

Overriding it for things if there's clumsiness (clown) does sound kinda funny

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -282,3 +282,38 @@
...
+
+		S = M.get_item_by_slot(slot_back)	//else we put in backpack
+		if(istype(S) && S.can_be_inserted(src,1))
+			S.handle_item_insertion(src)
+			playsound(src.loc, "rustle", 50, 1, -5)
+			return 1
+
+		M << "<span class='warning'>You are unable to equip that!</span>"
+		return 0
+	else return ..()
```
Comment (duncathan) [inline code/modules/reagents/reagent_containers/glass.dm]: This whole thing is copy paste how can you call this OOP

**Comment by Lo6a4evskiy** [comment]
Yes, but I don't have any other ideas on how to not have it equip to the assigned slot without doing that at the moment.

**Comment by tkdrg** [comment]
A better solution would be moving slot_equipment_priority to be an /obj/item var instead of a global var, and have each item change that, if needed. That way you won't need as much copy-paste.

**Comment by duncathan** [comment]
@Lo6a4evskiy

**Comment by Lo6a4evskiy** [comment]
I've seen it, end of semester however, got a shitton of other things to do practically every day. Will change it in a few days I think.

**Comment by tkdrg** [comment]
Sure, no rush.

**Comment by Lo6a4evskiy** [comment]
Ugh, I forgot to test the bucket itself and it doesn't work. Gotta fix it.

**Comment by tkdrg** [comment]
This looks much better now. No more copy-paste.

**Pull request merged**